r/R/dataset.R

```diff
@@ -64,6 +64,12 @@ open_dataset <- function(sources,
                          partitioning = hive_partition(),
                          unify_schemas = NULL,
                          ...) {
+  # If you aren't explicit with the argument names it looks like everything
+  # works but it will create a segfault and crash the R session, this fixes it
+  if (any(class(schema) %in% "HivePartitioning")) {
+    partitioning <- schema
+    schema <- NULL
+  }
```

Comment:
I see what you're going for here, and this is super helpful for our specific case! But I think it would be better to check that `schema` does `inherit()` from `"Schema"` and provide a helpful error message if it doesn't.

That way we also catch cases where  `schema` is passed some other object type and it's clear that the input was an error / there is a difference between the schema and partitioning